specs/_features/epbs/fork-choice.md

```diff
@@ -0,0 +1,634 @@
+    if header.parent_block_hash != parent_header.block_hash:
+        assert header.parent_block_hash == parent_header.parent_block_hash
+        return
+
+    # verify the inclusion list
+    assert verify_inclusion_list(state, block, inclusion_list, EXECUTION_ENGINE)
+    store.inclusion_list_available[root]=True
+```
+    
+### `notify_ptc_messages`
+
+```python
+def notify_ptc_messages(store: Store, state: BeaconState, payload_attestations: Sequence[PayloadAttestation]) -> None:
+    """
+    Extracts a list of ``PayloadAttestationMessage`` from ``payload_attestations`` and updates the store with them
+    These Payload attestations are assumed to be in the beacon block hence signature verification is not needed
+    """
+    if state.slot == 0:
+        return
+    for payload_attestation in payload_attestations:
+        indexed_payload_attestation = get_indexed_payload_attestation(state, state.slot - 1, payload_attestation)
+        for idx in indexed_payload_attestation.attesting_indices:
+            store.on_payload_attestation_message(PayloadAttestationMessage(validator_index=idx,
+                    data=payload_attestation.data, signature= BLSSignature(), is_from_block=true))
+```
+
+### `is_payload_present`
+
+```python
+def is_payload_present(store: Store, beacon_block_root: Root) -> bool:
+    """
+    Return whether the execution payload for the beacon block with root ``beacon_block_root`` was voted as present 
+    by the PTC
+    """
+    # The beacon block root must be known
+    assert beacon_block_root in store.ptc_vote
+    return store.ptc_vote[beacon_block_root].count(PAYLOAD_PRESENT) > PAYLOAD_TIMELY_THRESHOLD
+```
+
+### `is_parent_node_full`
+
+```python
+def is_parent_node_full(store: Store, block: BeaconBlock) -> bool:
+    parent = store.blocks[block.parent_root]
+    return block.body.signed_execution_payload_header.message.parent_block_hash == parent.body.signed_execution_payload_header.message.block_hash
+```
+
+### Modified `get_ancestor` 
+**Note:** `get_ancestor` is modified to return whether the chain is based on an *empty* or *full* block. 
+
+```python
+def get_ancestor(store: Store, root: Root, slot: Slot) -> ChildNode:
+    """
+    Returns the beacon block root, the slot and the payload status of the ancestor of the beacon block 
+    with ``root`` at ``slot``. If the beacon block with ``root`` is already at ``slot`` or we are 
+    requesting an ancestor "in the future" it returns its PTC status instead of the actual payload content. 
+    """
+    block = store.blocks[root]
+    if block.slot <= slot:
+        return ChildNode(root=root, slot=slot, is_payload_present=is_payload_present(store, root))
+
+    parent = store.blocks[block.parent_root]
+    if parent.slot > slot:
+        return get_ancestor(store, block.parent_root, slot)
+    return ChildNode(root=block.parent_root, slot=parent.slot, is_payload_present=is_parent_node_full(block))
```

Comment:
```suggestion
    return ChildNode(root=block.parent_root, slot=parent.slot, is_payload_present=is_parent_node_with_payload(block))
```

Reply:
can it be withheld or absent here?

Reply:
Both are equivalent from the point of view of this function